src/expr/src/expr/expr_regexp_count.rs

```diff
@@ -0,0 +1,252 @@
+                        let flags_datum = deserialize_datum(
+                            flags_value.get_body().as_slice(),
+                            &DataType::from(flags_node.get_return_type().unwrap()),
+                        )
+                        .map_err(|e| ExprError::Internal(e.into()))?;
+
+                        match flags_datum {
+                            Some(ScalarImpl::Utf8(flags)) => Some(flags.to_string()),
+                            _ => bail!("Expected flags to be a String"),
+                        }
+                    }
+                    _ => {
+                        return Err(ExprError::UnsupportedFunction(
+                            "non-constant flags in `regexp_count`".to_string(),
+                        ))
+                    }
+                }
+            }
+        };
+
+        // Sanity check
+        if children.next().is_some() {
+            bail!("syntax error in `regexp_count`");
+        }
+
+        let flags = if let Some(f) = flags {
+            f
+        } else {
+            "".to_string()
+        };
+
+        if flags.contains('g') {
+            bail!("`regexp_count` does not support global flag option");
+        }
+
+        let ctx = RegexpContext::new(&pattern, &flags)?;
+
+        Ok(Self {
+            source,
+            ctx,
+            return_type,
+            start,
+        })
+    }
+}
+
+impl RegexpCountExpression {
+    fn match_row(&self, text: Option<&str>) -> Option<i32> {
+        if let Some(text) = text {
+            // First get the start position to count for
+            let start = if let Some(s) = self.start { s - 1 } else { 0 };
+
+            // For unicode purpose
+            let mut start = match text.char_indices().nth(start as usize) {
+                Some((idx, _)) => idx,
+                // The start is out of bound or just invalid
+                // FIXME: What to return?
+                None => return None,
+            };
+
+            let mut count = 0;
+
+            while let Some(captures) = self.ctx.regex.captures(&text[start..]) {
+                count += 1;
+                start += captures.get(0).unwrap().end();
+            }
+
+            Some(count)
+        } else {
+            // Input string is None
+            // FIXME: Return None or Some(0)?
+            None
+        }
+    }
+}
+
+#[async_trait::async_trait]
+impl Expression for RegexpCountExpression {
+    fn return_type(&self) -> DataType {
+        self.return_type.clone()
+    }
+
+    async fn eval(&self, input: &DataChunk) -> Result<ArrayRef> {
+        let source_column = self.source.eval_checked(input).await?;
+        let source_column = source_column.as_utf8();
+
+        let row_len = input.capacity();
+        let vis = input.vis();
+        let mut builder: I32ArrayBuilder = ArrayBuilder::new(row_len);
+
+        for row_idx in 0..row_len {
+            if !vis.is_set(row_idx) {
+                builder.append_n(1, None);
```

Comment:
Simply `append` is okay.
```suggestion
                builder.append(None);
```

Reply:
I see, the `append` has the default implementation which could be directly used.